### PR TITLE
feature: fs cache refresh

### PR DESCRIPTION
**Pull request by dkhokhlov**
- added config.log.clear_cache_interval, default is 6 hours
- fs tailer gets restarted periodically using clear_cache_interval to clear fs cache 

LOG_15140

### CODE REVIEW

File: common/config/src/raw.rs [hunk 1]
```diff
@@ -445,6 +447,7 @@ impl Default for LogConfig {
             k8s_metadata_include: None,
             k8s_metadata_exclude: None,
             log_metric_server_stats: None,
+            clear_cache_interval: Some(3600 * 3), // 3 hours
```
Comment (stsantilena) [inline common/config/src/raw.rs:450]: Was 3 hours chosen at random or is there a significant build up over that period?

Comment (dkhokhlov) [reply common/config/src/raw.rs:450]: build up is unconstrained. depends on file event rate. by consensus.

File: common/fs/src/tail.rs [hunk 2]
```diff
@@ -319,6 +324,20 @@ where
...
+            if let Some(start_time) = this.start_time.as_mut().as_pin_mut() {
+                if this.restart_interval.as_secs() > 0
+                    && start_time.elapsed().as_secs() > this.restart_interval.as_secs()
+                {
+                    this.start_time.set(Some(Instant::now()));
+                    let stream_fut = (this.f)(this.params);
+                    this.pending.set(Some(stream_fut));
+                    info!(
+                        "restarting stream, interval={}",
+                        this.restart_interval.as_secs()
```
Comment (stsantilena) [inline common/fs/src/tail.rs:337]: Is that just resetting the interval which resets the loop?

**Review by stsantilena — APPROVED** [review]
Just had a couple small questions, other than that, this LGTM.

**Comment by dkhokhlov** [comment]
rebased to master, changed interval default to 6 hours.

**Pull request merged**